src/datastore/index.js

```diff
@@ -170,4 +170,26 @@ export class DataStore extends TypedEmitter {
     if (!block) throw new Error('Not Found')
     return decode(block, { coreDiscoveryKey, index })
   }
+
+  /** @param {Buffer} buf} */
+  async writeRaw(buf) {
+    const { length } = await this.#writerCore.append(buf)
+    const index = length - 1
+    const coreDiscoveryKey = this.#writerCore.discoveryKey
+    if (!coreDiscoveryKey) {
+      throw new Error('Writer core is not ready')
+    }
+    const versionId = getVersionId({ coreDiscoveryKey, index })
+    return versionId
+  }
+
+  /** @param {string} versionId */
+  async readRaw(versionId) {
+    const { coreDiscoveryKey, index } = parseVersionId(versionId)
+    const core = this.#coreManager.getCoreByDiscoveryKey(coreDiscoveryKey)
+    if (!core) throw new Error('Invalid versionId')
```

Comment:
This is actually a not found error, because the versionId parsed ok so it is valid.

Reply:
This would happen when a peer has not synced the core that has the data.